tests/lib/managers/files-test.js

```diff
@@ -2952,4 +2952,144 @@ describe('Files', function() {
+		var expectedRepresentationAllParams = { headers: { 'x-rep-hints': testRepresentation, set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png' }, qs: { fields: 'representations' } };
+		var placeholderUrl = '.../{+asset_path}';
+		var representationOptions = {asset_path: '1.png'};
+		var representationAllOptions = {asset_path: '1.png', set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png'};
+
+		it('should make GET request to get file representation when called', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationParam)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationOptions);
+		});
+
+		it('should make GET request to get file representation when called with all optional headers', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationAllParams)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationAllOptions);
+		});
+
+		it('should call callback with the representation when a 200 response is returned', function(done) {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [
+							{
+								content: {
+									url_template: placeholderUrl
+								}
+							}
+						]
+					}
+				}
+			};
+			sandbox.stub(boxClientFake, 'get').returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationOptions, function(err, representationObject) {
+				assert.ifError(err);
+				assert.strictEqual(representationObject, response.body.representations, 'representation object is returned');
+				done();
+			});
+		});
+		it('should return a promise resolving to a representation object when a 200 response is returned', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [
+							{
+								content: {
+									url_template: placeholderUrl
+								}
+							}
+						]
+					}
+				}
+			};
+			sandbox.stub(boxClientFake, 'get').returns(Promise.resolve(response));
+			return files.getRepresentationInfo(FILE_ID, testRepresentation, representationOptions)
+				.then(representationObject => {
+					assert.strictEqual(representationObject, response.body.representations, 'representation object is returned');
```

Comment:
Same as above — there should also be an assertion about the `asset_url` property here